src/averageAges.js

```diff
@@ -55,10 +69,25 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // 1. find a mother of each person (or only for men)
-  // 2. keep people who have mothers in the array
-  // 3. calculate the difference child.born - mother.born
-  // 4. return the average value
+  const filteredPeople = onlyWithSon
+    ? people.filter(person => person.sex === 'm'
+      && person.mother !== null)
+    : people.filter(person => person.mother !== null);
+
+  const ageDiffs = filteredPeople.map(person => {
+    const mother = people.find(p => p.name === person.mother);
+
+    if (mother && mother.born) {
+      return person.born - mother.born;
+    }
+
+    return null;
+  });
+
+  const validAgeDiffs = ageDiffs.filter(ageDiff => ageDiff !== null);
```

Comment:
looks like redundant. you can do it in reduce
in reduce you can check whether was found personMother and return sum + person.born - personMother.bore (if was found mother) or return sum;